File: THCS/2022_2023/TienGiang_2223/Bai1/SAME.py
def max_frequency(input_file, output_file):
    with open(input_file, 'r') as infile:
        n = int(infile.readline().strip())  # Đọc số lượng cây
        heights = list(map(int, infile.readline().strip().split()))  # Đọc chiều cao của các cây

    frequency = {}

    # Đếm tần số xuất hiện của từng chiều cao
    for height in heights:
        if height in frequency:
            frequency[height] += 1
        else:
            frequency[height] = 1

    # Tìm tần số lớn nhất
    max_freq = max(frequency.values()) if frequency else 0

    # Ghi kết quả vào tệp
    with open(output_file, 'w') as outfile:
        outfile.write(str(max_freq) + '\n')

# Gọi hàm với tên tệp đầu vào và đầu ra
max_frequency("SAME.INP", "SAME.OUT")

File: THCS/2022_2023/TienGiang_2223/Bai1/SAME.cpp
#include <iostream>
#include <fstream>
#include <unordered_map>
#include <vector>
using namespace std;

int main() {
    ifstream input("SAME.INP");
    ofstream output("SAME.OUT");

    int n;
    input >> n; // Đọc số lượng cây

    unordered_map<int, int> frequency; // Sử dụng unordered_map để đếm tần số
    vector<int> heights(n);

    // Đọc chiều cao của các cây và đếm tần số
    for (int i = 0; i < n; i++) {
        input >> heights[i];
        frequency[heights[i]]++;
    }

    // Tìm tần số lớn nhất
    int max_freq = 0;
    for (const auto& pair : frequency) {
        if (pair.second > max_freq) {
            max_freq = pair.second;
        }
    }

    // Ghi kết quả vào tệp
    output << max_freq << endl;

    input.close();
    output.close();
    return 0;
}
